xdsl/dialects/builtin.py

```diff
@@ -1175,8 +1175,18 @@ def from_strings(name: str, value: str, type: Attribute = NoneAttr()) -> OpaqueA
         return OpaqueAttr([StringAttr(name), StringAttr(value), type])
 
 
+class MemrefLayoutAttr(Attribute, ABC):
+    """
+    Interface for any attribute acceptable as a memref layout.
+    """
+
+    name = "memref_layout_att"
```

Comment:
Why provide a name?

Reply:
Oh no, this didn't just slip through :face_with_peeking_eye: 

This is a can of worm, and just putting a name here avoids to open it, can we separate the can of worm from this very practical PR?

But, to answer:
When using such a type in an IRDL constant, some verification errors try to access its name for the error reporting, erorring the error :upside_down_face:  (Also, it's defined at the `Attribute` baseclass level that they should all have a name, so Pyright says "okay" to that.) So there's a whole little discussion to have around that, but practically and currently, we need any base attribute used in Constraints to have a name. And per type consistency, they really just all should have a `name` - Pyright assumes so.

Reply:
Very interesting. Surely this is worth at least an issue? I think it would be great to have at least a bit of a discussion of some potential future.

Reply:
Definitely worth an issue, will file one in a moment if not done

And yeah, worth the discussion (or a type system 😂), I just really want this change independently in, as this is holding back a concrete project of our Belgium collaborators 🙂

Reply:
Thinking about it a little more, I'd recommend adding abstract in the name, filing an issue, and leaving the rest for later

Reply:
I'm okay with that!

Reply:
Oh god 🙈